### DISCUSSION TITLE
Customized GoPro Camera Functionality Integration

### DISCUSSION THREAD

**Discussion by Selvababu93** [gopro/labs#903] [General]
I'm reaching out to seek your expertise and insights on a project I'm currently working on involving GoPro cameras.Our project centers around enhancing user experiences at water parks by leveraging GoPro cameras on water slide rides. We're aiming to automate the recording process using field sensors strategically placed along the slides. Here are the main functionalities we're seeking to integrate:

1.Automated Triggering by Field Sensors: We aim to trigger the GoPro camera's recording functionality through strategically placed field sensors. Upon detecting a human crossing the sensor's path, the camera should start recording automatically. This automation is crucial for capturing dynamic moments seamlessly without manual intervention.
2.Still Capture in Certain Situations: Besides video recording, there are scenarios where we need the GoPro camera to capture still images. Similar to video recording, these still images should be triggered by field sensors and seamlessly transferred to a designated path for further processing.
3.Slow Motion Video Capture: In specific instances where slow-motion footage is desired, we plan to utilize the same field sensor-triggered mechanism to initiate slow-motion video capture. The resulting footage should be promptly transferred to our designated path for subsequent use.
4.Integration of Intro and Outro Clips: To enhance the visual appeal of the produced content, we aim to integrate intro and outro clips into the recorded footage. These clips will provide context and branding elements to the final output. Therefore, we're seeking support in seamlessly combining the recorded footage with the specified intro and outro clips.

We're keen to engage with the GitHub community to explore the feasibility of integrating these functionalities into GoPro cameras. If you have any insights, suggestions, or previous experiences with similar projects, we'd greatly appreciate your input.

Additionally, if you're aware of any existing tools, APIs, or resources that could facilitate this integration, we'd love to hear about them.

Thank you all for your time and support. We're excited to collaborate and learn from the community as we work towards enhancing user experiences with GoPro cameras.

**Comment by dnewman-gpsw** [comment]
If the camera are wired (USB) you need to explore Open GoPro.   If you intend hand users cameras that will be more complicated.  Still maybe doable.

**Reply by Selvababu93** [reply]
Hello @dnewman-gpsw 
Thank you for your response and suggestion! We appreciate your insight into exploring Open GoPro for wired camera setups.

In our project, we plan to primarily utilize hand-held GoPro cameras, which adds a layer of complexity as you've rightly pointed out. However, we're still eager to explore feasible solutions to automate the recording process and integrate additional functionalities.

While the hand-held aspect may present challenges, we believe that with the right approach and tools, it could still be achievable. We'll certainly look into Open GoPro as a potential avenue for our wired camera setups.

If you have any further recommendations or insights on how we can adapt our project to accommodate hand-held cameras, we'd greatly appreciate your input.

Thanks again for your valuable contribution to the discussion!

**Reply by dnewman-gpsw** [reply]
Open GoPro also supports BLE and WiFi.  I think you will likely need proximity BLE beacon triggers which is more of a Labs concept (not yet implemented).  You can also use GPS proximity today on HERO11.

**Reply by Selvababu93** [reply]
Hello @dnewman-gpsw 
after lot of R&D we reached the destination. while am testing with gopro13 am getting following error: SettingId.VIDEO_MODE does not contain a value 3
SettingId.FRAMING does not contain a value 100.
could you help to find out?

**Reply by dnewman-gpsw** [reply]
No idea.  Try a factory reset.

**Reply by dnewman-gpsw** [reply]
If those are Open GoPro errors, ask on that forum.
